src/lib/y2network/udev_rule.rb

```diff
@@ -160,6 +160,7 @@ def write_net_rules(udev_rules)
         Yast::SCR.Write(Yast::Path.new(".udev_persistent.rules"), udev_rules.map(&:to_s))
         # Writes changes to the rules file
         Yast::SCR.Write(Yast::Path.new(".udev_persistent.nil"), [])
+        Yast::SCR.UnmountAgent(Yast::Path.new(".udev_persistent"))
```

Comment:
What about defining some helper method for wrapping the agent access? Something like

```ruby

def use_udev_agent(&block)
  block.call
ensure
  Yast::SCR.UnmountAgent(Yast::Path.new(".udev_persistent"))
end

...

use_udev_agent do
  Yast::SCR.Write(Yast::Path.new(".udev_persistent.rules"), udev_rules.map(&:to_s))
  # Writes changes to the rules file
  Yast::SCR.Write(Yast::Path.new(".udev_persistent.nil"), [])
end
```

or we could additionally add some write wrapper

```ruby

def write_udev(&block)
  block.call
ensure
  Yast::SCR.Write(Yast::Path.new(".udev_persistent.nil"), [])
  Yast::SCR.UnmountAgent(Yast::Path.new(".udev_persistent"))
end
```

and simplify the `write_drivers_rules` and `write_net_rules` methods.

Reply:
Hmm, I think that's a case where abstraction complicates things. True, the UnmountAgent calls are repeated, but at 3 occurrences  it is acceptable. Furthermore, the SCR(udev_persistent) calls are all limited to this single source file. The abstraction would pay off if offered as external API, but not here.

Reply:
OK, let's keep the changes as minimal as possible...